base/libgit2/types.jl

```diff
@@ -670,32 +670,46 @@ function securezero!(cred::UserPasswordCredentials)
     return cred
 end
 
+function Base.:(==)(a::UserPasswordCredentials, b::UserPasswordCredentials)
+    a.user == b.user && a.pass == b.pass
+end
+
 "SSH credentials type"
 mutable struct SSHCredentials <: AbstractCredentials
     user::String
     pass::String
-    pubkey::String
     prvkey::String
+    pubkey::String
```

Comment:
what's the reasoning for rearranging the fields here? that could be breaking if anything external is constructing this type directly